pkg/scheduler/core/generic_scheduler.go

```diff
@@ -210,12 +210,16 @@ func (g *genericScheduler) selectHost(nodeScoreList framework.NodeScoreList) (st
 
 // numFeasibleNodesToFind returns the number of feasible nodes that once found, the scheduler stops
 // its search for more feasible nodes.
-func (g *genericScheduler) numFeasibleNodesToFind(numAllNodes int32) (numNodes int32) {
-	if numAllNodes < minFeasibleNodesToFind || g.percentageOfNodesToScore >= 100 {
+func (g *genericScheduler) numFeasibleNodesToFind(fwk framework.Framework, numAllNodes int32) (numNodes int32) {
+	adaptivePercentage := g.percentageOfNodesToScore
+	if p := fwk.PercentageOfNodesToScore(); p != 0 {
```

Comment:
this defaulting should probably happen in the v1beta1/defaults.go file; then we can just use the value as-is without the `p != 0` check. also helps to have all the defaulting in the same place and also in the future when we decide to remove the top-level field.

Reply:
+1, defaulting needs to happen in v1beta1/defaults.go. Basically when not set we default the value to the higher level field.